### PR TITLE
SOV-580: vertical tabs component

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.atlassian.net/browse/SOV-580

### CODE REVIEW

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  @apply pt-[2.875rem] w-full relative;
+}
+
+.tabs {
+  @apply w-full flex flex-col justify-center items-center gap-y-[2.5rem];
+}
+
+.content {
+  @apply w-full h-full p-[3.25rem] relative overflow-y-auto;
+  background-color: inherit;
```
Comment (pietro-maximoff) [inline packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css]:
```suggestion
  @apply w-full h-full p-[3.25rem] relative overflow-y-auto bg-inherit;
```

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
```
Comment (pietro-maximoff) [inline packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css]:
```suggestion
```

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.button {
+  @apply w-full block text-left text-gray-10 text-opacity-50 font-medium font-roboto;
+  &.active {
+    @apply text-opacity-100;
+  }
+
+  & p {
+    @apply text-base leading-[1.171875rem];
+
```
Comment (pietro-maximoff) [inline packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css]:
```suggestion
```

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    @apply block text-xs leading-[0.87875rem] mt-2;
+  }
+}
+
+
+.activeButtonIndicator {
+  --indicator-bg-color: #000;
+  @apply absolute right-0 w-0 h-0;
+  transform: translate(0, -1.7rem);
+  border-style: solid;
```
Comment (pietro-maximoff) [inline packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css]:
```suggestion
  @apply absolute right-0 w-0 h-0 border-solid;
  transform: translate(0, -1.7rem);
```

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import { VerticalTabsItemButtonProps } from '../VerticalTabs.types';
+import styles from './VerticalTabItem.module.css';
+
+export const VerticalTabItem: FC<VerticalTabsItemButtonProps> = ({
+  disabled,
+  label,
+  infoText,
+  active,
+  dataActionId,
```
Comment (pietro-maximoff) [inline packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.tsx]: i think makes sense to change it now dataActionId -> dataLayoutId and ping Alita or Luna to update test cases

**Review by pietro-maximoff — CHANGES_REQUESTED** [review]
need small fixes, in general, all great

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css (hunk 1 above)
Comment (Rickk137) [inline packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css]: bg-inherit

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+.header {
+  @apply pt-[2.875rem] pb-[4.5rem] w-full relative;
+}
+
+.footer {
+  @apply pt-[2.875rem] w-full relative;
+}
+
+.tabs {
+  @apply w-full flex flex-col justify-center items-center gap-y-[2.5rem];
```
Comment (Rickk137) [inline packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css]: gap-y-10

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.types.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  header?: (props: VerticalTabsProps) => ReactNode;
+  footer?: (props: VerticalTabsProps) => ReactNode;
+};
+
+export type VerticalTabsItem = {
+  label: ReactNode;
+  content: ReactNode;
+  infoText?: ReactNode;
+  disabled?: boolean;
+  dataActionId?: string;
```
Comment (Rickk137) [inline packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.types.ts]: could you change it to dataLayoutId

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css [hunk 8]
```diff
@@ -0,0 +1,28 @@
+.button {
+  @apply w-full block text-left text-gray-10 text-opacity-50 font-medium font-roboto;
+  &.active {
+    @apply text-opacity-100;
+  }
+
+  & p {
+    @apply text-base leading-[1.171875rem];
```
Comment (Rickk137) [inline packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css]: ``1.171875rem`` value is very custom. i think we should ask designer not to use it

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css [hunk 9]
```diff
@@ -0,0 +1,28 @@
+.button {
+  @apply w-full block text-left text-gray-10 text-opacity-50 font-medium font-roboto;
```
Comment (Rickk137) [inline packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css]: font-roboto is applied in via global class

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  onClick,
+}) => {
+  return (
+    <button
+      className={classNames(styles.button, { [styles.active]: active })}
+      disabled={disabled}
+      data-action-id={dataActionId}
```
Comment (Rickk137) [inline packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.tsx]: data-layout-id

**Review by Rickk137 — CHANGES_REQUESTED** [review]

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css (hunk 1 above)
Comment (creed-victor) [reply packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css]: removed entirely, will not be used.

File: packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css (hunk 4 above)
Comment (creed-victor) [reply packages/ui/src/2_molecules/VerticalTabs/components/VerticalTabItem.module.css]: removed entirely, will not be used.

File: packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css (hunk 1 above)
Comment (creed-victor) [reply packages/ui/src/2_molecules/VerticalTabs/VerticalTabs.module.css]: removed entirely, will not be used.

**Review by pietro-maximoff — APPROVED** [review]

**Comment by Rickk137** [comment]
@creed-victor Looks good. Only if you could add a on hover style to the VerticalTabItem

**Comment by Rickk137** [comment]
@creed-victor also there is a conflict

**Review by Rickk137 — APPROVED** [review]

**Review by LunaSov — APPROVED** [review]

**Pull request merged**